bigquery_etl/query_scheduling/dag_collection.py

```diff
@@ -121,6 +101,24 @@ def with_tasks(self, tasks):
 
         return self
 
+    def get_task_downstream_dependencies(self, task):
+        """Return all direct downstream dependencies of the task."""
+        # Cache the downstream dependencies for faster lookups.
+        if not hasattr(self, "_downstream_dependencies"):
+            downstream_dependencies = defaultdict(list)
+            for dag in self.dags:
+                for _task in dag.tasks:
+                    _task.with_upstream_dependencies(self)
+                    for upstream_dependency in (
+                        _task.depends_on + _task.upstream_dependencies
+                    ):
+                        downstream_dependencies[upstream_dependency.task_key].append(
+                            _task.to_ref(self)
+                        )
+            self._downstream_dependencies = downstream_dependencies
+
+        return self._downstream_dependencies[task.task_key]
```

Comment:
I changed this from a dictionary property to a method so that callers don't have to know about the implementation detail of what dictionary key is being used for the cache.